File: wsm-web/src/api/warehouse.ts
import request from './request'
import type { ApiResponse, PageParams, PageResult } from './request'

// ==================== 类型定义 ====================

export interface Warehouse {
  id: number
  code: string
  name: string
  address: string
  contact: string
  phone: string
  status: number
  warehouseType: string
  createdAt: string
}

export interface WarehouseArea {
  id: number
  warehouseId: number
  code: string
  name: string
  type: number
  status: number
  createdAt?: string
}

export interface WarehouseShelf {
  id: number
  warehouseId: number
  areaId: number
  code: string
  name: string
  categoryName: string
  status: number
  createdAt?: string
}

// ==================== 仓库 API ====================

export function getWarehouseList(params: PageParams & { keyword?: string; status?: number }) {
  return request.get<any, ApiResponse<PageResult<Warehouse>>>('/warehouses', { params })
}

export function createWarehouse(data: Partial<Warehouse>) {
  return request.post<any, ApiResponse<number>>('/warehouses', data)
}

export function updateWarehouse(data: Partial<Warehouse>) {
  return request.put<any, ApiResponse<void>>('/warehouses', data)
}

export function deleteWarehouse(id: number) {
  return request.delete<any, ApiResponse<void>>(`/warehouses/${id}`)
}

// ==================== 库区 API ====================

export function getAreaList(warehouseId: number) {
  return request.get<any, ApiResponse<WarehouseArea[]>>(`/warehouse-areas/warehouse/${warehouseId}`)
}

export function createArea(data: Partial<WarehouseArea>) {
  return request.post<any, ApiResponse<number>>('/warehouse-areas', data)
}

export function updateArea(data: Partial<WarehouseArea>) {
  return request.put<any, ApiResponse<void>>('/warehouse-areas', data)
}

export function deleteArea(id: number) {
  return request.delete<any, ApiResponse<void>>(`/warehouse-areas/${id}`)
}

// ==================== 货架 API ====================

export function getShelfList(warehouseId: number) {
  return request.get<any, ApiResponse<WarehouseShelf[]>>(`/warehouse-shelves/warehouse/${warehouseId}`)
}

export function createShelf(data: Partial<WarehouseShelf>) {
  return request.post<any, ApiResponse<number>>('/warehouse-shelves', data)
}

export function updateShelf(data: Partial<WarehouseShelf>) {
  return request.put<any, ApiResponse<void>>('/warehouse-shelves', data)
}

export function deleteShelf(id: number) {
  return request.delete<any, ApiResponse<void>>(`/warehouse-shelves/${id}`)
}

File: wms-backend/src/main/java/com/yiweilai/wms/warehouse/entity/Warehouse.java
package com.yiweilai.wms.warehouse.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 仓库实体
 */
@Data
public class Warehouse {

    /** 仓库ID */
    private Long id;

    /** 仓库编码 */
    private String code;

    /** 仓库名称 */
    private String name;

    /** 仓库地址 */
    private String address;

    /** 联系人 */
    private String contact;

    /** 联系电话 */
    private String phone;

    /** 状态：1-启用 0-禁用 */
    private Integer status;

    /** 仓库类型：NORMAL-普通仓 DEFECTIVE-次品仓 SCRAP-报废仓 */
    private String warehouseType;

    /** 逻辑删除 */
    private Integer deleted;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}
